enhancements/kube-apiserver/stability-csrs.md

```diff
@@ -0,0 +1,202 @@
+---
+title: stability-csr-state
+authors:
+  - "@deads2k"
+reviewers:
+  - TBD
+  - "@alicedoe"
+approvers:
+  - TBD
+  - "@oscardoe"
+creation-date: yyyy-mm-dd
+last-updated: yyyy-mm-dd
+status: provisional|implementable|implemented|deferred|rejected|withdrawn|replaced
+see-also:
+  - "/enhancements/this-other-neat-thing.md"  
+replaces:
+  - "/enhancements/that-less-than-great-idea.md"
+superseded-by:
+  - "/enhancements/our-past-effort.md"
+---
+
+# Stability: CSR state exposure
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Graduation criteria for dev preview, tech preview, GA
+- [ ] User-facing documentation is created in [openshift-docs](https://github.com/openshift/openshift-docs/)
+
+## Open Questions [optional]
+
+This is where to call out areas of the design that require closure before deciding
+to implement the design.  For instance, 
+ > 1. This requires exposing previously private resources which contain sensitive
+  information.  Can we do this? 
+
+## Summary
+
+Having CSRs in certain states can risk cluster-death.
+Existing alerts don't fire during failed installation
+Existing alerts are suppressable and in many cases, suppressing them risks cluster-death.
+If your cluster dies because of this, you may require manual intervention for recovery. 
+
+## Motivation
+
+We often see UPI and metal clusters fail to install with this failure mode.
+We've had at least on kube-controller-manager failure on this in 4.x. 
+
+### Goals
+
+1. Having more than X kubelet CSRs unapproved for more than Y minutes is worthy of a degraded condition because it risks
+   cluster-death or indicates an attack.
+   The machine-api should know the difference in IPI flows.
+   On UPI flows, we should default a more generic degraded message.
+2. Having more than X kubelet CSRs approved, but not issued for than Y minutes is worthy of a degraded condition because
+   it risks cluster-death and indicates a misbehaving kube-controller-manager-operator.
+
+### Non-Goals
+
+1. Try to determine why the CSRs are unapproved.
+
+## Proposal
+
+This is a straight-forward count from machine-api-operator and kube-controller-manager-operator.
```

Comment:
machine-api-operator does not handle CSRs.  These are handled by the cluster-machine-approver.